m4/mpi.m4

```diff
@@ -226,50 +226,44 @@ AS_IF([test "x$MPI_IMPL" != x],
-                                [ dnl it gives true if MPI_VERSION !=1...
-                                 dnl check that MPI_VERSION is not 2; else through a warning
-                                 AC_TRY_LINK([@%:@include <mpi.h>],
-                                             [
-                                             #if MPI_VERSION == 2
-                                             #error "deprecated MPI-version"
-                                             #endif
-                                             int np; MPI_Comm_size (MPI_COMM_WORLD, &np)
-                                             ],
-                                             [
-                                               dnl MPI_VERSION >2: Be happy and configure with MPI.
-                                              AC_MSG_RESULT([ The MPI found has version >= 3.0. ]);
-                                             ],
-                                             [
-                                              AC_MSG_WARN([MPI-version is 2.X. This feature is deprecated. ]);
-                                             ])
-                                dnl set necessary variables etc: Here, all enablempi-cases are in one place.
-                                AC_DEFINE(HAVE_MPI, 1, [Flag indicating whether or not MPI is available])
-                                ],
-                                [
-                                 AC_MSG_WARN(["ERROR: MPI-version is 1.X. Currently we support only MPI version >= 2.0.  Disable MPI now..."]); enablempi=no
-                                ])
+                     dnl if MPI_VERSION is 2, through a warning
+                     AC_TRY_LINK([@%:@include <mpi.h>],
+                                 [@%:@if MPI_VERSION == 2
+                                 @%:@error "deprecated MPI-version"
+                                 @%:@endif
+                                 int np; MPI_Comm_size (MPI_COMM_WORLD, &np)
+                                 ],
+                                 [
+                                    AC_MSG_RESULT([ The MPI found has version >= 3.0. ]);
+                                 ],
+                                 [
+                                    AC_MSG_WARN([MPI-version is 2.X. This feature is deprecated. ]);
+                                 ])
+                    dnl set necessary variables etc: Here, all enablempi-cases are in one place.
+                    AC_DEFINE(HAVE_MPI, 1, [Flag indicating whether or not MPI is available])
                     ],
-                    [ dnl MPI_VERSION is not defined.
-                     AC_MSG_WARN(["ERROR: MPI-version seems to be too low: MPI < 1.5. Disable MPI now..."]); enablempi=no
+                    [ dnl MPI_VERSION is not defined or has unexpected value.
+                     AC_MSG_WARN(["ERROR: MPI-version seems to be too low: Need MPI 2.X or 3.X. Disable MPI now..."]); enablempi=no
```

Comment:
I'd say "2.x or compatible" - that'll be accurate whether a fully-backwards-compatible or deprecated-stuff-removed releases come out in the future.